ui-speedspacechart/src/components/helpers/drawElements/speedLimitTags.ts

```diff
@@ -0,0 +1,185 @@
+  marginTop: number,
+  store: Store
+): Promise<tooltipInfos | null> => {
+  const {
+    speedLimitTags,
+    ratioX,
+    leftOffset,
+    layersDisplay: { electricalProfiles, powerRestrictions },
+    cursor,
+  } = store;
+
+  const { MARGIN_TOP, MARGIN_BOTTOM, MARGIN_LEFT, MARGIN_RIGHT } = MARGINS;
+
+  if (!speedLimitTags) return null;
+  clearCanvas(ctx, width, LINEAR_LAYERS_HEIGHTS.SPEED_LIMIT_TAGS_HEIGHT);
+
+  ctx.save();
+  ctx.translate(leftOffset, 0);
+
+  const { maxPosition } = maxPositionValues(store);
+
+  const questionImage = await loadSvgImage(questionBlobUrl);
+  const alertFillImage = await loadSvgImage(alertFillBlobUrl);
+
+  let speedLimitTagsBackgroundColor = LINEAR_LAYERS_BACKGROUND_COLOR.FIRST;
+
+  if (electricalProfiles && powerRestrictions) {
+    speedLimitTagsBackgroundColor = LINEAR_LAYERS_BACKGROUND_COLOR.THIRD;
+  } else if (electricalProfiles || powerRestrictions) {
+    speedLimitTagsBackgroundColor = LINEAR_LAYERS_BACKGROUND_COLOR.SECOND;
+  }
+
+  drawSeparatorLinearLayer(ctx, 'rgba(0,0,0,0.1)', MARGINS, width, 1);
+  drawLinearLayerBackground(
+    ctx,
+    speedLimitTagsBackgroundColor,
+    MARGINS,
+    width,
+    MARGIN_BOTTOM,
+    LINEAR_LAYERS_HEIGHTS.SPEED_LIMIT_TAGS_HEIGHT - 1
+  );
+
+  const RECT_HEIGHT = 17;
+  const yPosition = 12;
+  const RECTANGLE_SPACING = 1;
+  const TEXT_LEFT_PADDING = 4;
+  const TEXT_RIGHT_PADDING = 8;
+  const FIRST_TAG_LEFT_PADDING = 8;
+  const ICON_WIDTH = 16;
+  const ICON_HEIGHT = 16;
+  const ICON_OFFSET = 4;
+  const TEXT_PADDING_TOP = 1;
+
+  let tooltip: tooltipInfos | null = null;
+
+  speedLimitTags.boundaries.forEach((boundary, index) => {
+    const tag = speedLimitTags.values[index];
+    if (tag && tag.speed_limit_tags_type === 'tag') {
+      const x = positionOnGraphScale(boundary, maxPosition, width, ratioX, MARGINS);
+      const nextBoundary = speedLimitTags.boundaries[index + 1];
+
+      if (nextBoundary !== undefined) {
+        const tagWidth =
+          positionOnGraphScale(nextBoundary, maxPosition, width, ratioX, MARGINS) -
+          x -
+          RECTANGLE_SPACING;
+
+        ctx.fillStyle = tag.color;
+        ctx.strokeStyle = tag.color;
+        ctx.lineWidth = 1;
+
+        ctx.beginPath();
+        ctx.rect(x, yPosition, tagWidth - RECTANGLE_SPACING, RECT_HEIGHT);
+        ctx.closePath();
+        ctx.fillRect(x, yPosition, tagWidth - RECTANGLE_SPACING, RECT_HEIGHT);
+
+        ctx.fillStyle = tag.color;
+        const textWidth =
+          ctx.measureText(tag.tag_name).width +
+          TEXT_RIGHT_PADDING +
+          (index === 0 ? FIRST_TAG_LEFT_PADDING : 0);
+
+        if (tag.color === '#216482') {
+          ctx.fillStyle = '#E5F7FF';
+        } else if (tag.color === '#D91C1C') {
+          ctx.fillStyle = '#F15981';
+          ctx.strokeStyle = '#494641';
+        } else if (tag.color === '#494641') {
+          ctx.fillStyle = '#F2F0E4';
+        } else {
+          ctx.fillStyle = tag.color;
+        }
```

Comment:
I think this not should be hardcoded, you should have dictionary to associate colors and use tag.color as a key to find the association

Reply:
@clarani we can keep this improvement in another PR ?